### ISSUE TITLE
groupby documentation is incorrect

### ISSUE THREAD

**Issue by fetis** [mozilla/nunjucks#430]
Current documentation leads to http://jinja.pocoo.org/docs/dev/templates/#groupby
But result of groupby is object like
```
{
groupKeyValue1: [],
groupKeyValue2: []
}
```

So, first method with `group.grouper` and `group.list` doesn't work.
Second method works fine.

**Comment by jlongster** [comment]
I think this is a difference in Python's destructuring.

It looks like Python's `groupby` filter returns data like this: `[ { grouper: 'key1', list: [...] }, { grouper: 'key2', list: [...] } ]`. If I remember correctly, Python's destructuring in `for k, v in lst` will see that `lst` is an array, iterate over it, and destructure each item into `k, v`. So it just kind of "works" that `k` is bound to the group key and `v` is bound to the list.

That's why you can also just iterate over the list and get an object with `grouper` and `list` properties.

JavaScript does not have any magic destructuring when iterating. You have to explicitly destructure. The equivalent in JS would be `for { grouper: k, list: v } of arr`. This is one of those places where we embrace the fact that we are a JS engine, and we aren't going to try and emulate everything about Python, even if it is nicer.

Basically we can only support one of these interfaces. I'd be open to adding an additional argument to `groupby` to return data in a different shape so that you can do `group.grouper`, etc

**Comment by fetis** [comment]
I'm absolutely ok with this behavior. I just want that documentation reflects supported API. 
Is it ok if I create page for filters and put description of `groupby` there? We can keep other links on jinja2.

**Comment by jlongster** [comment]
Ok! If you wait until https://github.com/mozilla/nunjucks/pull/428 lands, you'll see that I started documenting out filters. I'd love it if you added `groupby` in there too, I just started a few of them.
